loki/transformations/argument_shape.py

```diff
@@ -80,6 +82,35 @@ def transform_subroutine(self, routine, **kwargs):  # pylint: disable=arguments-
                                          if d == ':']
                             vmap[arg] = arg.clone(type=arg.type.clone(shape=new_shape))
 
+                    elif re.search(r'\*$', str(arg.shape[-1])):
+                        # determine extent of explicit shape part of assumed size declaration
+                        expl_shape_len = [loc for loc, dim in enumerate(arg.shape) if '*' in dim][0]
```

Comment:
The standard explicitly mandates `expl_shape_len = len(arg.shape)-1`, can't we use that knowledge here?

> The rank is equal to one plus the number of explicit-shape-specs.
https://j3-fortran.org/doc/year/10/10-007.pdf#page=114